db/migrate/20241204221534_add_indexes_to_saved_claims.rb

```diff
@@ -0,0 +1,8 @@
+class AddIndexesToSavedClaims < ActiveRecord::Migration[7.1]
+  disable_ddl_transaction!
+
+  def change
+    add_index :saved_claims, [:metadata, :type], where: "(metadata LIKE '%error%')", algorithm: :concurrently
```

Comment:
I'm a little nervous on this index. Every insert into this table is going to have to do a fuzzy search of the `metadata` field. If that field is big and/or there are a lot of inserts into this table that might cause a spike in CPU/insert time. Do we have any other options?

Reply:
Here's a list of the options I considered:
1) partial index `metadata LIKE '%error%'` as implemented in this PR - metadata is `NULL` on INSERTs and gets populated later by our status jobs
2) GIN index on `metadata` - this one is theoretically the best option if we are doing arbitrary LIKE queries, but for our purposes we only check if `error` is present which is why I went with the partial index. To my understanding, GIN indexes are very fast but use a lot of storage.
```
add_index :saved_claims, :metadata, algorithm: :concurrently, using: :gin, opclass: :gin_trgm_ops
```
3) add a new column for (error) status which is updated by the status updater jobs, and add an index on the status column - this one is probably the easiest/most performant way if we don't mind making additions to the table that aren't likely to be used by other teams.

Reply:
Thanks @dfong-adh for this! I think you made a good decision with the partial index as a middle ground. Approved!